### PR TITLE
Allow Temporary Accommodation to be developed on top of Approved Premises

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Context

[This PR](https://trello.com/c/oC1y8aIe/395-spike-adding-a-property-to-the-frontend) allows us to develop Temporary Accommodation service, side-by-side with the Approved Premises service, sharing code where possible.

This PR doesn't go as far as #229 with namespacing parts of the code where we want TA and AP to diverge, since we expect `temporary-accommodation-main` to be a long-running branch, and moving a lot of files would be make later rebases painful

This PR adds several new environment variables. See `example.env`, and as a starting point, copy these new variables into your `.env` file

# Changes in this PR

* All existing AP functionality is now accessibly locally at `http://localhost:3000/approved-premises`
* There is a new stub TA page at `http://localhost:3000/temporary-accommodation`

## Screenshots of UI changes

localhost_3000_

### CODE REVIEW

File: server/utils/applicationUtils.test.ts
Comment (tahb) [inline server/utils/applicationUtils.test.ts:44]: This line could possibly be removed in favour of the same option set a few lines below: 

```
config.temporaryAccommodationSubdomain = 'temporary-accommodation-subdomain'
```

Which looks to have the right value assigned to it too.

File: server/utils/applicationUtils.ts
Comment (tahb) [inline server/utils/applicationUtils.ts:25]: Non-blocking suggestion: There's a fair bit of conditional logic going on in one method, there might be an opportunity to refactor this into some private methods. That said it's readable and unlikely to get changed much more. This approach may be more common in the Typescript world in which case I'll get back in my hole :)

File: server/utils/applicationUtils.test.ts
Comment (tahb) [inline server/utils/applicationUtils.test.ts:41]: Excellent to see the code and tests going into the same commit, love to see it 👍

File: server/routes/temporary-accommodation/index.ts
Comment (tahb) [inline server/routes/temporary-accommodation/index.ts:1]: Today I learned. An instruction to exclude this file from test coverage?

**Review by tahb — APPROVED** [review]
I've read through as best I can though lots is unfamiliar to me. I note that Stu has already had a look at the earlier draft ([229](https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/229)) you pointed towards and there wasn't any lower level feedback. I wonder if that's because it's all good or because Stu didn't have the time to give it more than a high level review? If it's the latter we might want to get a review from someone more useful than me. Given the convergence work here that affects AP I suggest it'd be worth confirming that with Stu as we want this to be good enough to be merged into `main` when ready?

In general though, as I said, I love seeing lots of test coverage and small commits that help me follow along. 

I'll approve from my end and let you have that final check with Stu before merging.

File: server/paths/api.ts
Comment (pezholio) [inline server/paths/api.ts:1]: I think this is a really good idea - combining application and API paths always make me a bit uneasy

File: server/controllers/index.ts
Comment (pezholio) [inline server/controllers/index.ts:5]: I wondere if there's a more elegant way of namespacing the controllers? What I'd like to see is a single set of "standard" controllers in the base path, and then a way of overriding the standard controllers by adding a controller to a namespaced path, and the same with views. That said, I don't think this is a blocker given you're working on a feature branch, but something worth thinking about.

File: server/config.ts
Comment (pezholio) [inline server/config.ts:90]: I wonder if we should have default service signifiers, so locally we're using the path and then in 'prod-like' envs we're using the domain

File: server/routes/temporary-accommodation/index.ts
Comment (bryangaledxw) [reply server/routes/temporary-accommodation/index.ts:1]: Seems to be - this is following the convention of the AP routes

File: server/utils/nunjucksSetup.ts
Comment (pezholio) [inline server/utils/nunjucksSetup.ts:40]: We should probably call the apps just `Approved Premises` or `Temporary Accommodation`. The UI distinction is irrelevant. I've been meaning to change this for ages, was just a default name that got put in when we generated the repo from the template

**Review by pezholio — APPROVED** [review]
I think this is all fine, nice and easy to follow. No blockers given this is going into a feature branch, but a few suggestions / things to think about

File: server/utils/applicationUtils.test.ts
Comment (bryangaledxw) [reply server/utils/applicationUtils.test.ts:44]: Ah, this should be

```
config.temporaryAccommodation*RootPath* = 'temporary-accommodation-root-path'
```

Will fix

**Pull request merged**
